src/datachain/lib/utils.py

```diff
@@ -28,3 +30,29 @@ def __init__(self, message):
 class DataChainColumnError(DataChainParamsError):
     def __init__(self, col_name, msg):
         super().__init__(f"Error for column {col_name}: {msg}")
+
+
+def normalize_col_names(col_names: Sequence[str]) -> dict[str, str]:
+    gen_col_counter = 0
+    new_col_names = {}
+    org_col_names = set(col_names)
+
+    for org_column in col_names:
+        new_column = org_column.lower()
+        new_column = re.sub("[-_\\s]+", "_", new_column)
+        new_column = new_column.strip("_")
+        new_column = re.sub("[^0-9a-z_]+", "", new_column)
+
+        if not new_column or (new_column != org_column and new_column in org_col_names):
+            while True:
+                generated_column = f"c{gen_col_counter}"
+                gen_col_counter += 1
+                if new_column:
+                    generated_column = f"{generated_column}_{new_column}"
+                if generated_column not in org_col_names:
+                    new_column = generated_column
+                    break
```

Comment:
```suggestion
        if not new_column or (new_column != org_column and new_column in new_col_names):
            while True:
                generated_column = f"c{gen_col_counter}"
                gen_col_counter += 1
                if new_column:
                    generated_column = f"{generated_column}_{new_column}"
                if generated_column not in new_col_names:
                    new_column = generated_column
                    break
```

I thought the whole point of this logic was checking for duplicates in the `new_column_names` dictionary?

### Test vector

```python
normalize_col_names(("column+name", "column=name"))
```

#### Before
```python
assert result == {"columnname": "column=name"}
```

#### After
```python
assert result == {"columnname": "column+name", "c0_columnname": "column=name"}
```

Reply:
sorry about that, I'm fixing all of this right now ... give me a few minutes a review again pls

Reply:
> I thought the whole point of this logic was checking for duplicates in the new_column_names dictionary?

it's not that simple ... we should also ideally check that there is no already a column in the original list and try to avoid naming it the same if possible